Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -43,7 +43,9 @@ final class PurchasesOrchestrator {
 
     private let _allowSharingAppStoreAccount: Atomic<Bool?> = nil
     private let presentedOfferingIDsByProductID: Atomic<[String: String]> = .init([:])
-    private let purchaseCompleteCallbacksByProductID: Atomic<[String: PurchaseCompletedBlock]> = .init([:])
+    private let purchaseCompleteCallbacksByProductID: Atomic<[
+        String: (completion: PurchaseCompletedBlock, date: Date)
```

Comment:
maybe we should have this typed into a struct rather than a tuple? that'd be a little more expressive + if we ever need to add anything more it's easier to add

Reply:
also maybe we can name this `callbackCreationDate` instead of `date` for clarity, otherwise you might think this is the date of the corresponding transaction or something

Reply:
Both good suggestions 👍🏻  will do.